src/ConnectedMode/UI/ManageConnections/ManageConnectionsViewModel.cs

```diff
@@ -23,162 +23,164 @@
 using SonarLint.VisualStudio.ConnectedMode.UI.Resources;
 using SonarLint.VisualStudio.Core.WPF;
 
-namespace SonarLint.VisualStudio.ConnectedMode.UI.ManageConnections
+namespace SonarLint.VisualStudio.ConnectedMode.UI.ManageConnections;
+
+internal class ManageConnectionsViewModel(
+    IConnectedModeUIManager connectedModeUiManager,
+    IConnectedModeServices connectedModeServices,
+    IConnectedModeBindingServices connectedModeBindingServices,
+    IProgressReporterViewModel progressReporterViewModel) : ViewModelBase
 {
-    internal class ManageConnectionsViewModel(
-        IConnectedModeServices connectedModeServices,
-        IConnectedModeBindingServices connectedModeBindingServices,
-        IProgressReporterViewModel progressReporterViewModel) : ViewModelBase
+    public IConnectedModeUIManager ConnectedModeUiManager { get; } = connectedModeUiManager;
+    public IConnectedModeBindingServices ConnectedModeBindingServices { get; } = connectedModeBindingServices;
+    public IProgressReporterViewModel ProgressReporterViewModel { get; } = progressReporterViewModel;
+    public ObservableCollection<ConnectionViewModel> ConnectionViewModels { get; } = [];
+    public bool NoConnectionExists => ConnectionViewModels.Count == 0;
+
+    internal async Task LoadConnectionsWithProgressAsync()
     {
-        public IConnectedModeBindingServices ConnectedModeBindingServices { get; } = connectedModeBindingServices;
-        public IProgressReporterViewModel ProgressReporterViewModel { get; } = progressReporterViewModel;
-        public ObservableCollection<ConnectionViewModel> ConnectionViewModels { get; } = [];
-        public bool NoConnectionExists => ConnectionViewModels.Count == 0;
+        var validationParams = new TaskToPerformParams<ResponseStatus>(
+            InitializeConnectionViewModelsAsync,
+            UiResources.LoadingConnectionsText,
+            UiResources.LoadingConnectionsFailedText);
+        await ProgressReporterViewModel.ExecuteTaskWithProgressAsync(validationParams);
+    }
 
-        internal async Task LoadConnectionsWithProgressAsync()
-        {
-            var validationParams = new TaskToPerformParams<ResponseStatus>(
-                async () => await SafeExecuteActionAsync(InitializeConnectionViewModels),
-                UiResources.LoadingConnectionsText,
-                UiResources.LoadingConnectionsFailedText);
-            await ProgressReporterViewModel.ExecuteTaskWithProgressAsync(validationParams);
-        }
+    internal async Task RemoveConnectionWithProgressAsync(List<string> bindingKeysReferencedByConnection, ConnectionViewModel connectionViewModel)
+    {
+        var validationParams = new TaskToPerformParams<ResponseStatus>(
+            async () => await RemoveConnectionViewModelAsync(bindingKeysReferencedByConnection, connectionViewModel),
```

Comment:
Similar to the `CreateNewConnectionAsync` comment
```suggestion
            () => Task.FromResult(RemoveConnectionViewModel(bindingKeysReferencedByConnection, connectionViewModel)),
```

Reply:
I prefer the await syntax better and let the responsibility of returning a Task to the `RemoveConnectionViewModelAsync`.

Reply:
By design this method is not async, it's not its responsibility to return a Task. The Task is only needed because of the `TaskToPerformParams`, if the `TaskToPerformParams` wasn't calling the `RemoveConnectionViewModel`, then the `RemoveConnectionViewModel` wouldn't make async.

Adding the async is making the `RemoveConnectionViewModel`'s body use unnecessary `await Task.FromResult`.

Reply:
Sorry just noticed that the `CreateNewConnectionAsync` comment was never posted. It's a similar case though, no need to be async, neither to return a Task.